### ISSUE TITLE
MotorTrend On Demand

### ISSUE THREAD

**Issue by Veritas413** [ytdl-org/youtube-dl#25580]
## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2020.06.06**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single video: https://www.motortrendondemand.com/detail/the-650hp-clevo-commando-returns/153906/
- Site: https://www.motortrendondemand.com/

## Description

Used to be able to download just fine from https://www.motortrendondemand.com/ - looks like they changed something to break it, or changed their backend hosting.\
Tried with login and cookies, which used to work, but now it doesn't.
Found a closed request from last year (https://github.com/ytdl-org/youtube-dl/issues/23317) but the solution doesn't seem to work anymore, and the video IDs from the URL changed format.
In addition, it looks like they're now requiring logins. I can provide to someone working on the issue, if they don't want to sign up for a free trial that requires a CC.

**Comment by ualnwe** [comment]
Seems they've switched from a Kaltura backend to Discovery's own backend: `disco`/`dplay`?

Should be able to adapt an existing disco/dplay extractor to work, and it seems the auth/token flow is similar too.

**Comment by cjwil** [comment]
I'm not sure how helpful this is to devs but hopefully it will be a helpful workaround until its fixed.

In Firefox if you open the inspect element on a video and switch to the network tab, you may have to refresh the page but sort the information by type and then copy that file's url and paste that into youtube-dl and it should work. I've tested it a few times and a couple videos and it seems to be working like the old way did.

**Comment by Veritas413** [comment]
> I'm not sure how helpful this is to devs but hopefully it will be a helpful workaround until its fixed.
> 
> In Firefox if you open the inspect element on a video and switch to the network tab, you may have to refresh the page but sort the information by type and find the one that says dash in the type column. Then copy that file's url and paste that into youtube-dl and it should work. I've tested it a few times and a couple videos and it seems to be working like the old way did.

Nice! That appears to be a sufficient workaround! It's a pain, but it's better than nothing.

**Comment by cjwil** [comment]
Adding to this again because it seems they've made more changes. If when searching for the URL in the inspector, the file name is only "Master.mpd" and the URL contains "&KeyName=primary&Signature" you must put the URL in quotes for youtube-dl or the & symbols mess with the URL

**Comment by ufcpride** [comment]
Does the workaround work for all videos? i used to never have an issue with the site, but since they changed i get the same. I have a paid account subscription, but as above i like to pull these down and view on Plex.  I wasn't able to find a master.mpd or dash url. Here is the one i'm trying. https://www.motortrendondemand.com/detail/easy-does-it/1118741/

**Comment by dgcurtis** [comment]
Also, if you want the subtitles/captions, you can search for "Mastercaption".  If you right-click and get the curl link, you can paste that into your shell and redirect the output to a vtt file.  Then you can merge the video and vtt with ffmpeg.

**Comment by JamesRokas** [comment]
Hey i know this is an old thread, but does anybody have a suggestion on parameters to use for any motortrend content so it keeps the file names?

**Comment by cjwil** [comment]
I'm not sure if it's possible to do that with youtube-dl due to the way that MotorTrend serves their videos but I use Sonarr for that.

**Comment by JamesRokas** [comment]
Yeah i just kind of gave up and i’ve been naming them all myself and grabbing them one by one. it takes a while but worth it i guess. i’ll look into sonar though! i’ve heard really good things about it but haven’t had the time to get into it

**Comment by cjwil** [comment]
You do have the option of copy and pasting the name into the terminal as well like youtube-dl -o "NAME" "URL" however symbols like exclamation points can mess this up

**Comment by JamesRokas** [comment]
most of them do have exclamation marks in the name so i’ll just keep doing what i’m doing hahaha. thank you though. i really appreciate it!

**Comment by xxtensazenxx** [comment]
Sorry to re-open this thread but downloading the .mpd file does not work anymore, it just downloads the ad. Anyone have this working still in 2022?

**Comment by Veritas413** [comment]
Still working for me.
Firefox, CTRL+Shift+I, Network tab, refresh page, start video, pause video.
Sort by type or filter to find an octet-stream called master.mpd - right click, copy URL
Paste into youtube-dl or yt-dlp, using [-o "WhateverYouWantTheVideoToBeCalled.mp4"]

**Comment by dgcurtis** [comment]
Yeah, still works here too. I keep hoping the extractor gets fixed but since it's been a few years, it's not looking hopeful.
